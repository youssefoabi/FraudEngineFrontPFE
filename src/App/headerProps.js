import { Link } from 'react-router-dom';

export default {
  navigationProps: {
    Link,
    entries: [
      {
        label: 'SERVICES',
        entries: [
          {
            label: 'PARAMETRAGE',
            entries: [
              {
                label: 'Moteurs de recherche',
                path: '/engines',
              },
              {
                label: 'Groupe de moteurs de recherche',
                path: '/engines-groups',
              },
              {
                label: 'Référentiel des gammes',
                path: '/scopes',
              },
            ],
          },
        ],
      },
    ],
  },
};
