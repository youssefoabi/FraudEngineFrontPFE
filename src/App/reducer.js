import { combineReducers } from 'redux';

import { enginesEntities, enginesNavigation } from '../components/EnginesDataGrid/reducers';
import { navigationEngineError, navigationEngineSuccess } from '../components/EngineForm/reducers';
import {
  navigationEnginesGroupError,
  navigationEnginesGroupSuccess,
} from '../components/EnginesGroupForm/reducers';
import { navigationScopeError, navigationScopeSuccess } from '../components/ScopeForm/reducers';
import { scopesEntities, scopesNavigation } from '../components/ScopesDataGrid/reducers';
import pendingActionNavigation from '../components/ServerActionBackdrop/reducers/navigationReducer';

const rootReducer = combineReducers({
  entities: combineReducers({
    scopes: scopesEntities,
    engines: enginesEntities,
  }),
  navigation: combineReducers({
    scopes: scopesNavigation,
    scopeFormError: navigationScopeError,
    scopeFormSuccess: navigationScopeSuccess,
    engineFormError: navigationEngineError,
    engineFormSuccess: navigationEngineSuccess,
    enginesGroupFormError: navigationEnginesGroupError,
    enginesGroupFormSuccess: navigationEnginesGroupSuccess,
    pendingRequest: pendingActionNavigation,
    engines: enginesNavigation,
  }),
});

export default rootReducer;
