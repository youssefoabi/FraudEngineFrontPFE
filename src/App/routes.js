import CriteriaManagement from '../screens/CriteriaManagement';
import CriterionCreation from '../screens/CriterionCreation';
import CriterionEdition from '../screens/CriterionEdition';
import Home from '../screens/Home';
import RuleCreation from '../screens/RuleCreation';
import RuleEdition from '../screens/RuleEdition';
import RulesManagement from '../screens/RulesManagement';

export default [
  {
    exact: true,
    path: '/',
    component: Home,
    name: 'Accueil',
  },
  {
    exact: true,
    path: '/rules/create',
    component: RuleCreation,
    name: "Création d'une règle",
  },
  {
    exact: true,
    path: '/rules/edit/:id',
    component: RuleEdition,
    name: "Edition d'une règle",
  },
  {
    exact: true,
    path: '/rules',
    component: RulesManagement,
    name: 'Gestion des règles',
  },
  {
    exact: true,
    path: '/criteria/create',
    component: CriterionCreation,
    name: "Création d'un critère",
  },
  {
    exact: true,
    path: '/criteria/edit/:id',
    component: CriterionEdition,
    name: "Edition d'un critère",
  },
  {
    exact: true,
    path: '/criteria',
    component: CriteriaManagement,
    name: 'Gestion des critères',
  },
];
