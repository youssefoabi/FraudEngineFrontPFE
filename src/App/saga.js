import { all } from 'redux-saga/effects';

import engineCreationOperations from '../screens/EngineCreation/operations';
import engineEditionOperations from '../screens/EngineEdition/operations';
import enginesGroupCreationOperations from '../screens/EnginesGroupCreation/operations';
import enginesGroupEditionOperations from '../screens/EnginesGroupEdition/operations';
import enginesManagementOperations from '../screens/EnginesManagement/operations';
import scopeCreationOperations from '../screens/ScopeCreation/operations';
import scopeEditionOperations from '../screens/ScopeEdition/operations';
import scopesManagementOperations from '../screens/ScopesManagement/operations';

export default function* watchAll() {
  yield all([...scopeCreationOperations]);
  yield all([...scopeEditionOperations]);
  yield all([...scopesManagementOperations]);
  yield all([...engineCreationOperations]);
  yield all([...engineEditionOperations]);
  yield all([...enginesManagementOperations]);
  yield all([...enginesGroupCreationOperations]);
  yield all([...enginesGroupEditionOperations]);
}
