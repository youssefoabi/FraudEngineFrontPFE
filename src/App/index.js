import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { Core } from '@cdiscount/backoffice-app-components';
import { Header, Layout, ThemeProvider } from '@cdiscount/backoffice-ui';
import React from 'react';

import headerProps from './headerProps';
import rootReducer from './reducer';
import rootSaga from './saga';
import routes from './routes';

function App() {
  return (
    <Core reducer={rootReducer} saga={rootSaga}>
      <ThemeProvider>
        <BrowserRouter>
          <Layout>
            <Header {...headerProps} />
            <Switch>
              {routes.map(({ component, exact, path }) => (
                <Route key={path} component={component} exact={exact} path={path} />
              ))}
            </Switch>
          </Layout>
        </BrowserRouter>
      </ThemeProvider>
    </Core>
  );
}

export default App;
