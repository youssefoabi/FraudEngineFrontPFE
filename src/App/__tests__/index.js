import { itRendersAllMutations } from '@cdiscount/jest-utils';

import App from '../';

describe(App.name, () => {
  itRendersAllMutations(App, [
    {
      name: 'with default props',
      props: {},
    },
  ]);
});
