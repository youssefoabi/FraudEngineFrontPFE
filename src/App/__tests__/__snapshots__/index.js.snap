// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders correctly with default props 1`] = `
<Core
  middlewares={Array []}
  preloadedState={Object {}}
  reducer={[Function]}
  saga={[Function]}
>
  <c
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {
          "MuiButton": Object {
            "contained": Object {
              "boxShadow": "none",
            },
            "label": Object {
              "fontSize": "0.625rem",
              "fontWeight": 700,
              "textTransform": "uppercase",
            },
            "outlined": Object {
              "backgroundColor": "#fff",
            },
            "outlinedPrimary": Object {
              "&:hover": Object {
                "borderWidth": 2,
              },
              "borderWidth": 2,
            },
            "outlinedSecondary": Object {
              "&:hover": Object {
                "borderWidth": 2,
              },
              "borderWidth": 2,
            },
            "root": Object {
              "borderRadius": 6,
              "boxShadow": "none",
              "minHeight": 30,
            },
            "sizeLarge": Object {
              "minHeight": "2.125rem",
            },
            "sizeSmall": Object {
              "minHeight": "1.625rem",
            },
          },
          "MuiDialogActions": Object {
            "root": Object {
              "backgroundColor": "#CAD5E4",
              "padding": 10,
            },
          },
          "MuiDialogContent": Object {
            "root": Object {
              "padding": 20,
            },
          },
          "MuiDialogTitle": Object {
            "root": Object {
              "backgroundColor": "#CAD5E4",
              "padding": 10,
            },
          },
          "MuiFormControlLabel": Object {
            "label": Object {
              "fontSize": 11,
            },
          },
          "MuiFormLabel": Object {
            "root": Object {
              "color": "#2A3847",
              "fontSize": 11,
              "fontWeight": 800,
              "marginLeft": 5,
              "textTransform": "uppercase",
            },
          },
          "MuiIcon": Object {
            "fontSizeLarge": Object {
              "fontSize": "1.25rem",
            },
            "fontSizeSmall": Object {
              "fontSize": "0.75rem",
            },
            "root": Object {
              "fontSize": "1rem",
            },
          },
          "MuiInput": Object {
            "input": Object {
              "&:focus": Object {
                "boxShadow": "0 0 0 0.03rem #CAD5E4",
              },
              "backgroundColor": "#fff",
              "border": "2px solid #CAD5E4",
              "borderRadius": 6,
              "color": "#2A3847",
              "fontSize": 11,
              "height": "auto",
              "padding": "6px 0 7px 10px",
            },
            "root": Object {
              "padding": 0,
            },
          },
          "MuiInputAdornment": Object {
            "positionEnd": Object {
              "marginLeft": 0,
            },
            "positionStart": Object {
              "marginRight": 0,
            },
            "root": Object {
              "position": "absolute",
              "right": 5,
            },
          },
          "MuiInputLabel": Object {
            "root": Object {
              "color": "#2A3847",
              "fontSize": 11,
              "fontWeight": 800,
              "marginLeft": 5,
              "textTransform": "uppercase",
            },
          },
          "MuiPaper": Object {
            "rounded": Object {
              "borderRadius": 6,
            },
          },
          "MuiSelect": Object {
            "icon": Object {
              "color": "#2A3847",
              "height": "100%",
              "paddingLeft": 8,
              "paddingRight": 9,
              "top": 0,
            },
            "root": Object {
              "backgroundColor": "#fff",
              "borderRadius": 6,
              "borderWidth": 2,
            },
            "select": Object {
              "&:focus": Object {
                "borderRadius": 6,
                "boxShadow": "none",
              },
              "borderRadius": 6,
            },
            "selectMenu": Object {
              "paddingRight": 36,
            },
          },
          "MuiSvgIcon": Object {
            "fontSizeLarge": Object {
              "fontSize": "1.25rem",
            },
            "fontSizeSmall": Object {
              "fontSize": "0.75rem",
            },
            "root": Object {
              "fontSize": "1rem",
            },
          },
          "MuiTableCell": Object {
            "head": Object {
              "& span, & span:hover, & span:focus": Object {
                "color": "#fff",
              },
              "color": "#fff",
              "fontWeight": 700,
            },
            "paddingCheckbox": Object {
              "padding": "0 12px",
            },
            "root": Object {
              "border": "none",
              "borderWidth": 0,
              "fontSize": 11,
              "padding": "4px 56px 4px 24px",
            },
          },
          "MuiTablePagination": Object {
            "select": Object {
              "&.MuiSelect-select": Object {
                "paddingLeft": 10,
                "paddingRight": 45,
              },
              "&:focus": Object {
                "borderRadius": 6,
                "boxShadow": "none",
              },
              "border": "2px solid #3EC2A7",
              "borderRadius": 6,
              "color": "#3EC2A7",
              "fontWeight": 700,
            },
            "selectIcon": Object {
              "borderLeft": "1px solid #3EC2A7",
              "color": "#3EC2A7",
              "top": 0,
            },
            "selectRoot": Object {
              "color": "#3EC2A7",
            },
            "toolbar": Object {
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          },
          "MuiTableRow": Object {
            "head": Object {
              "backgroundColor": "#1B2731 !important",
              "height": 36,
              "textTransform": "uppercase",
            },
            "root": Object {
              "&:nth-child(even)": Object {
                "backgroundColor": "#DFE5EF",
              },
              "&:nth-child(odd)": Object {
                "backgroundColor": "#CAD5E4",
              },
              "height": 48,
            },
          },
          "MuiTableSortLabel": Object {
            "icon": Object {
              "fill": "#fff",
            },
          },
          "MuiToolbar": Object {
            "root": Object {
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          },
        },
        "palette": Object {
          "action": Object {
            "activatedOpacity": 0.12,
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "disabledOpacity": 0.38,
            "focus": "rgba(0, 0, 0, 0.12)",
            "focusOpacity": 0.12,
            "hover": "rgba(0, 0, 0, 0.04)",
            "hoverOpacity": 0.04,
            "selected": "rgba(0, 0, 0, 0.08)",
            "selectedOpacity": 0.08,
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#fafafa",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#000",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "rgb(158, 40, 3)",
            "light": "rgb(231, 97, 55)",
            "main": "#E23A05",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "info": Object {
            "contrastText": "#fff",
            "dark": "#1976d2",
            "light": "#64b5f6",
            "main": "#2196f3",
          },
          "primary": Object {
            "contrastText": "#fff",
            "dark": "#1B2731",
            "light": "#2A3847",
            "main": "#2A3847",
          },
          "quaternary": Object {
            "contrastText": "#2A3847",
            "dark": "#DFE5EF",
            "light": "#E9EEF4",
            "main": "#CAD5E4",
          },
          "quinary": Object {
            "contrastText": "#000",
            "dark": "#000",
            "light": "#CAD5E4",
            "main": "#fff",
          },
          "secondary": Object {
            "contrastText": "#fff",
            "dark": "rgb(43, 135, 116)",
            "light": "#CAD5E4",
            "main": "#3EC2A7",
          },
          "success": Object {
            "contrastText": "#fff",
            "dark": "rgb(80, 143, 21)",
            "light": "rgb(143, 215, 75)",
            "main": "#73CD1F",
          },
          "tertiary": Object {
            "contrastText": "#fff",
            "main": "#00B8FF",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
          "warning": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "#f57c00",
            "light": "#ffb74d",
            "main": "#ff9800",
          },
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
          "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
          "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
          "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
          "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
          "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
          "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
          "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
          "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
          "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
          "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
          "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
          "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
          "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
          "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
          "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
          "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
          "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
        ],
        "shape": Object {
          "borderRadius": 4,
        },
        "spacing": [Function],
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": 1.5,
          },
          "body2": Object {
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "lineHeight": 1.43,
          },
          "button": Object {
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "lineHeight": 1.75,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "lineHeight": 1.66,
          },
          "fontFamily": "\\"Open Sans\\", sans-serif",
          "fontSize": 14,
          "fontWeightBold": 700,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "h1": Object {
            "@media (min-width:1280px)": Object {
              "fontSize": "5.9983rem",
            },
            "@media (min-width:600px)": Object {
              "fontSize": "4.7129rem",
            },
            "@media (min-width:960px)": Object {
              "fontSize": "5.3556rem",
            },
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "3.5rem",
            "fontWeight": 300,
            "lineHeight": 1.167,
          },
          "h2": Object {
            "@media (min-width:1280px)": Object {
              "fontSize": "3.75rem",
            },
            "@media (min-width:600px)": Object {
              "fontSize": "2.9167rem",
            },
            "@media (min-width:960px)": Object {
              "fontSize": "3.3333rem",
            },
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "2.375rem",
            "fontWeight": 300,
            "lineHeight": 1.2,
          },
          "h3": Object {
            "@media (min-width:1280px)": Object {
              "fontSize": "2.9991rem",
            },
            "@media (min-width:600px)": Object {
              "fontSize": "2.5707rem",
            },
            "@media (min-width:960px)": Object {
              "fontSize": "2.7849rem",
            },
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "2rem",
            "fontWeight": 400,
            "lineHeight": 1.167,
          },
          "h4": Object {
            "@media (min-width:1280px)": Object {
              "fontSize": "2.0243rem",
            },
            "@media (min-width:600px)": Object {
              "fontSize": "1.8219rem",
            },
            "@media (min-width:960px)": Object {
              "fontSize": "2.0243rem",
            },
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "1.5625rem",
            "fontWeight": 400,
            "lineHeight": 1.235,
          },
          "h5": Object {
            "@media (min-width:1280px)": Object {
              "fontSize": "1.4993rem",
            },
            "@media (min-width:600px)": Object {
              "fontSize": "1.3118rem",
            },
            "@media (min-width:960px)": Object {
              "fontSize": "1.4993rem",
            },
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "1.25rem",
            "fontWeight": 400,
            "lineHeight": 1.334,
          },
          "h6": Object {
            "@media (min-width:1280px)": Object {
              "fontSize": "1.25rem",
            },
            "@media (min-width:600px)": Object {
              "fontSize": "1.25rem",
            },
            "@media (min-width:960px)": Object {
              "fontSize": "1.25rem",
            },
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "1.125rem",
            "fontWeight": 500,
            "lineHeight": 1.6,
          },
          "htmlFontSize": 16,
          "overline": Object {
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "lineHeight": 2.66,
            "textTransform": "uppercase",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subtitle1": Object {
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "lineHeight": 1.75,
          },
          "subtitle2": Object {
            "fontFamily": "\\"Open Sans\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "lineHeight": 1.57,
          },
          "useNextVariants": true,
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "speedDial": 1050,
          "tooltip": 1500,
        },
      }
    }
  >
    <BrowserRouter>
      <l>
        <p
          actions={Array []}
          navigationProps={
            Object {
              "Link": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Link",
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "entries": Array [
                Object {
                  "entries": Array [
                    Object {
                      "entries": Array [
                        Object {
                          "label": "Moteurs de recherche",
                          "path": "/engines",
                        },
                        Object {
                          "label": "Groupe de moteurs de recherche",
                          "path": "/engines-groups",
                        },
                        Object {
                          "label": "Référentiel des gammes",
                          "path": "/scopes",
                        },
                      ],
                      "label": "PARAMETRAGE",
                    },
                  ],
                  "label": "SERVICES",
                },
              ],
            }
          }
        />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            key="/"
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/scopes/create"
            path="/scopes/create"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/scopes/edit/:id"
            path="/scopes/edit/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/scopes"
            path="/scopes"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/engines/create"
            path="/engines/create"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/engines/edit/:id"
            path="/engines/edit/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/engines"
            path="/engines"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/engines-groups/create"
            path="/engines-groups/create"
          />
          <Route
            component={[Function]}
            exact={true}
            key="/engines-groups/edit/:id"
            path="/engines-groups/edit/:id"
          />
        </Switch>
      </l>
    </BrowserRouter>
  </c>
</Core>
`;
